assignment/assignment-tool/tool/src/webapp/js/assignments.js

```diff
@@ -362,29 +362,29 @@ ASN.showOrHideSelectGroupsMessage = function() {
     }
     
     // Get the form submission buttons
-    var btnPost = document.getElementById("post");
-    var btnSave = document.getElementById("save");
-    var btnPreview = document.getElementById("preview");
-    var buttons = [btnPost, btnPreview];
-    if (btnSave !== null) {
-        buttons.push(btnSave);
-    }
-        
+    var postButtons = document.getElementsByName( "post" );
+    var previewButtons = document.getElementsByName( "preview" );
+    var saveButtons = document.getElementsByName( "save" );
+
     // Show/hide the groups message
     if (groupsRadio.checked && !groupsSelected) {
         groupMsg.style.display = "block";
         
         // Disable the post, save and preview buttons
-        for (i = 0; i < buttons.length; i++) {
-            buttons[i].disabled = true;
+        for (i = 0; i < postButtons.length; i++) {
```

Comment:
I was trying to avoid having three separate loops, but maybe that's the best option. I'll refactor.